torch/_dynamo/symbolic_convert.py

```diff
@@ -2030,6 +2030,22 @@ def check_inlineable(func):
                 f"call torch._dynamo.disable() wrapped function {func.get_function()}"
             )
 
+        if isinstance(func, NestedUserFunctionVariable) and func.closure is not None:
+            # closure vars other than 'self' are not in scope of generated code, so error early
+            # TODO(avik): these should be supported eventually
+            closure_vars = [
+                var.name
+                for var in func.closure.items
+                if isinstance(var, ClosureVariable) and var.name != "self"]
+            if closure_vars:
+                code = func.get_code()
+                UserError(
+                    UserErrorType.ANTI_PATTERN,
+                    f"Cannot inline nested function '{code.co_name}' at {code.co_filename}:{code.co_firstlineno} "
+                    f"because it closes over variables {closure_vars}. "
+                    f"Please rewrite '{code.co_name}' to take {closure_vars} as additional arguments."
```

Comment:
When it's a missing feature, I like adding a link to a github issue where people can ask us to prioritize adding the feature

Reply:
Good idea.